### PR TITLE
Login command

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes: https://github.com/dart-lang/pub/issues/2467

### CODE REVIEW

File: lib/src/command/login.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      }
+      log.warning('You are already logged in as $userInfo\n'
+          'Run `pub logout` to log out and try again.');
+    }
+  }
+
+  Future<_UserInfo> retrieveUserInfo() async {
+    return await oauth2.withClient(cache, (client) async {
+      final discovery = await httpClient
+          .get('https://accounts.google.com/.well-known/openid-configuration');
```
Comment (jonasfj) [inline lib/src/command/login.dart:44]: I know that `https://www.googleapis.com/oauth2/v1/userinfo` can return a user profile too.. but I can't find documentation for the API end-point :see_no_evil:

File: lib/src/command/login.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    });
+  }
+}
+
+class _UserInfo {
+  final String name;
+  final String email;
+  _UserInfo(this.name, this.email);
+  @override
+  String toString() => '$name($email)';
```
Comment (jonasfj) [inline lib/src/command/login.dart:63]: `<$email> "$name"`

Comment (sigurdm) [inline lib/src/command/login.dart:63]:
```suggestion
  String toString() => '<$email> "$name"';
```

**Comment by stefandevo** [comment]
Maybe this is something that can be added specifically for the official repository pub.dev?
https://github.com/dart-lang/pub/pull/2654 has added login too, but for hosted repository only.

**Pull request merged**